src/main/scala/org/broadinstitute/dsde/firecloud/service/NihService.scala

```diff
@@ -72,78 +74,88 @@ trait NihService extends LazyLogging {
   val thurloeDao: ThurloeDAO
   val googleDao: GoogleServicesDAO
 
+  val nihWhitelists: Set[NihWhitelist] = FireCloudConfig.Nih.whitelists
+
   def getNihStatus(userInfo: UserInfo): Future[PerRequestMessage] = {
     thurloeDao.getProfile(userInfo) flatMap {
       case Some(profile) =>
         profile.linkedNihUsername match {
           case Some(_) =>
-            rawlsDao.isDbGapAuthorized(userInfo) map { isDbGapAuthorized =>
-              RequestComplete(NihStatus(profile.copy(linkExpireTime = Option(profile.linkExpireTime.getOrElse(0L))), Some(isDbGapAuthorized)))
+            Future.traverse(nihWhitelists) { whitelistDef =>
+              rawlsDao.isGroupMember(userInfo, whitelistDef.groupToSync).map(isMember => NihWhitelistStatus(whitelistDef.name, isMember))
+            }.map { whitelistMembership =>
+              RequestComplete(NihStatus(profile.copy(linkExpireTime = Option(profile.linkExpireTime.getOrElse(0L))), whitelistMembership))
             }
           case None => Future.successful(RequestComplete(NotFound))
         }
       case None => Future.successful(RequestComplete(NotFound))
     }
   }
 
-  def updateNihLinkAndSyncSelf(userInfo: UserInfo, nihLink: NIHLink): Future[PerRequestMessage] = {
-    val syncWhiteListResult = syncWhitelistUser(userInfo.getUniqueId, nihLink.linkedNihUsername)
-    val profilePropertyMap = nihLink.propertyValueMap
-    val propertyUpdates = thurloeDao.saveKeyValues(userInfo, profilePropertyMap)
+  private def downloadNihWhitelist(whitelist: NihWhitelist): Set[String] = {
+    val usersList = Source.fromInputStream(googleDao.getBucketObjectAsInputStream(FireCloudConfig.Nih.whitelistBucket, whitelist.fileName))
 
-    val profileResponse = propertyUpdates.map { response =>
-      if(response.isSuccess) RequestComplete(OK)
-      else RequestCompleteWithErrorReport(InternalServerError, "Error updating NIH link.")
-    }
-
-    syncWhiteListResult flatMap { _ => profileResponse } recover { case t => RequestCompleteWithErrorReport(InternalServerError, "Error updating NIH link", t) }
+    usersList.getLines().toSet
   }
 
-  private def getWhitelist(): Set[String] = {
-    val (bucket, file) = (FireCloudConfig.Nih.whitelistBucket, FireCloudConfig.Nih.whitelistFile)
-    val whitelist = Source.fromInputStream(googleDao.getBucketObjectAsInputStream(bucket, file))
+  // This syncs all of the whitelists in full
+  def syncNihWhitelistsFull: Future[PerRequestMessage] = {
```

Comment:
I prefer the `AllUsers` name.  It's not clear what a non-full sync would be, but `AllUsers` is an obvious contrast to a single user sync.